core/src/main/java/io/questdb/cairo/TableWriter.java

```diff
@@ -1441,57 +1441,115 @@ public boolean removePartition(long timestamp) {
+        // up metadata with logical partition delete we have to uphold the effort and
+        // re-compute table size and its minTimestamp from what remains on disk
 
-        // find out if we are removing min partition
-        long nextMinTimestamp = minTimestamp;
-        if (timestamp == txWriter.getPartitionTimestamp(0)) {
-            nextMinTimestamp = readMinTimestamp(txWriter.getPartitionTimestamp(1));
-        }
-        long partitionNameTxn = txWriter.getPartitionNameTxnByPartitionTimestamp(timestamp);
-        columnVersionWriter.removePartition(timestamp);
+        if (timestamp == getPartitionLo(maxTimestamp)) {
+            // removing the active partition
+            rollback();
 
-        txWriter.beginPartitionSizeUpdate();
-        txWriter.removeAttachedPartitions(timestamp);
-        txWriter.setMinTimestamp(nextMinTimestamp);
-        txWriter.finishPartitionSizeUpdate(nextMinTimestamp, txWriter.getMaxTimestamp());
-        txWriter.bumpTruncateVersion();
+            // calculate new boundaries
+            final long openTimestamp;
+            final long newFixedRowCount;
+            final long newTransientRowCount;
+            final long newMaxTimestamp;
+            final long newMinTimestamp;
+            if (index > 0) {
+                final int prevIndex = index - 1;
 
-        columnVersionWriter.commit();
-        txWriter.setColumnVersion(columnVersionWriter.getVersion());
-        txWriter.commit(defaultCommitMode, denseSymbolMapWriters);
+                openTimestamp = txWriter.getPartitionTimestamp(prevIndex);
+                newTransientRowCount = txWriter.getPartitionSize(prevIndex);
+                newFixedRowCount = txWriter.getFixedRowCount() - newTransientRowCount;
 
-        // Call O3 methods to remove check TxnScoreboard and remove partition directly
-        safeDeletePartitionDir(timestamp, partitionNameTxn);
+                setPathForPartition(path.trimTo(rootLen), partitionBy, openTimestamp, false);
+                TableUtils.txnPartitionConditionally(path, txWriter.getPartitionNameTxn(prevIndex));
+                readPartitionMinMax(ff, openTimestamp, path, metadata.getColumnName(metadata.getTimestampIndex()), newTransientRowCount);
+                newMinTimestamp = Math.max(attachMinTimestamp, openTimestamp);
+                newMaxTimestamp = Math.min(attachMaxTimestamp, partitionCeilMethod.ceil(openTimestamp));
+            } else {
+                // we are dropping the only partition
+                openTimestamp = 0L;
+                newFixedRowCount = 0L;
+                newTransientRowCount = 0L;
+                newMinTimestamp = Long.MAX_VALUE;
+                newMaxTimestamp = Long.MIN_VALUE;
+            }
+
+            // are there any readers, if so will remove the partition async-ly
+            boolean deletePartitionFolderAsync;
+            final long txn = txWriter.getTxn();
```

Comment:
You don't need to figure if the delete of partition be async here. It's too early to do before commit